crates/sargon/src/wrapped_radix_engine_toolkit/high_level/manifest_building/account_locker/manifest_account_locker.rs

```diff
@@ -0,0 +1,151 @@
+use crate::prelude::*;
+use radix_engine_interface::blueprints::locker::{
+    ACCOUNT_LOCKER_CLAIM_IDENT, ACCOUNT_LOCKER_CLAIM_NON_FUNGIBLES_IDENT,
+};
+
+impl From<AccountAddress> for ScryptoComponentAddress {
+    fn from(value: AccountAddress) -> ScryptoComponentAddress {
+        ScryptoComponentAddress::new_or_panic(value.node_id().0)
+    }
+}
+
+impl TransactionManifest {
+    pub fn account_locker_claim(
+        locker_address: &LockerAddress,
+        claimant: &AccountAddress,
+        claimable_resources: Vec<AccountLockerClaimableResource>,
+    ) -> Self {
+        let mut builder = ScryptoManifestBuilder::new();
+        let claimant_arg: ScryptoComponentAddress = (*claimant).into();
+        let claimable_resources =
+            Self::build_claimable_batch(claimable_resources, 50);
+
+        for claimable in claimable_resources.clone() {
+            match claimable {
+                AccountLockerClaimableResource::Fungible {
+                    resource_address,
+                    amount,
+                } => {
+                    let resource_arg: ScryptoResourceAddress =
+                        resource_address.into();
+                    let amount_arg: ScryptoDecimal192 = amount.into();
+
+                    builder = builder.call_method(
+                        locker_address,
+                        ACCOUNT_LOCKER_CLAIM_IDENT,
+                        (claimant_arg, resource_arg, amount_arg),
+                    );
+                }
+                AccountLockerClaimableResource::NonFungible {
+                    resource_address,
+                    ids,
+                } => {
+                    let resource_arg: ScryptoResourceAddress =
+                        resource_address.into();
+                    let ids_arg: Vec<ScryptoNonFungibleLocalId> =
+                        ids.iter().map(|id| id.clone().into()).collect();
+
+                    builder = builder.call_method(
+                        locker_address,
+                        ACCOUNT_LOCKER_CLAIM_NON_FUNGIBLES_IDENT,
+                        (claimant_arg, resource_arg, ids_arg),
+                    );
+                }
+            }
+        }
+
+        if !claimable_resources.is_empty() {
+            builder = builder.deposit_batch(claimant_arg);
```

Comment:
hmm, should we always use `deposit_batch`?
For asset transfers we do have explicit logic in place that determines if `deposit_batch` or `try_deposit_batch_or_abort` should be used depending on the account deposit config. I think we should do the same here.

Reply:
Well spotted!